pennylane/transforms/specs.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for resource estimation"""
+
+
+def specs(qnode, max_expansion=None):
+    """
+
+    Args:
+        qnode (.QNode): the QNode to calculation the specifications for
+
+    Keyword Args:
+        max_expansion=None (int): The number of times the internal circuit should be expanded when
+            executed on a device. Expansion occurs when an operation or measurement is not
+            supported, and results in a gate decomposition. If any operations in the decomposition
+            remain unsupported by the device, another expansion occurs. Defaults to
+            ``qnode.max_expansion``.
```

Comment:
```suggestion
            calculating the specification. Defaults to ``qnode.max_expansion``.
```
Since now this is uncoupled from executing on the device?